zerver/lib/push_notifications.py

```diff
@@ -603,6 +603,20 @@ def get_apns_alert_subtitle(message: Message) -> str:
     # For group PMs, or regular messages to a stream, just use a colon to indicate this is the sender.
     return message.sender.full_name + ":"
 
+def get_apns_badge_count(user_profile: UserProfile, read_messages_ids: Optional[List[int]]=[]) -> int:
+    """
+    If we have any messages that are read on the server, we don't
+    add their count even if the `active_mobile_push_notification`
+    flag is set. So, they will be excluded while getting the count.
+    """
```

Comment:
This doesn't really explain *why* we do this; it just repeats *what* the code says.

Here's an explanation of the "why":

```python
    # ...
    ).exclude(
        # If we've just marked some messages as read, they're still
        # marked as having active notifications; we'll clear that flag
        # only after we've sent that update to the devices.  So we need
        # to exclude them explicitly from the count.
        message_id__in=read_messages_ids
    ).count()
```

Note also that it's a code comment, not a docstring. Docstrings are appropriate for explaining the interface of a function; this is really about its implementation, which is better as a comment.